### PR TITLE
NUnit Issue 4686: Add ProgressTraceListener

### PR DESCRIPTION
**Pull request by maettu-this**
Docs PR to https://github.com/nunit/nunit/pull/4709:
VS Test Adapter documentation on Trace and Debug moved to NUnit and extended with ProgressTraceListener.

### CODE REVIEW

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (OsirisTerje) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:1]: You can't delete this file.  Either keep it, or make sure it links to the new one.  Somebody out there may have links to this page.

**Review by OsirisTerje — DISMISSED** [review]

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (maettu-this) [reply docs/articles/vs-test-adapter/Trace-and-Debug.md:1]: Sure, done with https://github.com/nunit/docs/pull/932/commits/f688cf1936db7426d763dc63ef00f5360e460de4.

**Review by SeanKilleen — APPROVED** [review]

File: docs/articles/nunit/technical-notes/usage/Trace-and-Debug-Output.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        {
+            Debug.WriteLine("This is Debug.WriteLine");
+            Trace.WriteLine("This is Trace.WriteLine");
+            Console.WriteLine("This is Console.Writeline");
+            TestContext.WriteLine("This is TestContext.WriteLine");
+            TestContext.Out.WriteLine("This is TestContext.Out.WriteLine");
+            TestContext.Progress.WriteLine("This is TestContext.Progress.WriteLine");
+            TestContext.Error.WriteLine("This is TestContext.Error.WriteLine");
+            Assert.Pass();
+        }
```
Comment (SeanKilleen) [inline docs/articles/nunit/technical-notes/usage/Trace-and-Debug-Output.md:59]:
```suggestion
[Test]
public void Test1()
{
    Debug.WriteLine("This is Debug.WriteLine");
    Trace.WriteLine("This is Trace.WriteLine");
    Console.WriteLine("This is Console.Writeline");
    TestContext.WriteLine("This is TestContext.WriteLine");
    TestContext.Out.WriteLine("This is TestContext.Out.WriteLine");
    TestContext.Progress.WriteLine("This is TestContext.Progress.WriteLine");
    TestContext.Error.WriteLine("This is TestContext.Error.WriteLine");
    Assert.Pass();
}
```

nitpick: No need for the indent here. I'll merge it 👍

**Comment by SeanKilleen** [comment]
@maettu-this thank you for this contribution! 🎉  Happy to get it merged.

**Comment by maettu-this** [comment]
@SeanKilleen, just wanted to confirm you are aware that the related feature https://github.com/nunit/nunit/pull/4709 still hasn't been merged yet, i.e. the wiki currently is ahead of the implemented mainline functionality. Having seen you and @OsirisTerje having conversation further above made me assume it's just a question of time (e.g. waiting for the fixed 4.1.0 release) before the feature PR gets looked at. But nothing has been happening in the meantime.

**Comment by OsirisTerje** [comment]
@maettu-this The build for https://github.com/nunit/nunit/pull/4709 fails. Have you looked at that?

**Comment by maettu-this** [comment]
@OsirisTerje, yes, as mentioned at https://github.com/nunit/nunit/pull/4709#issuecomment-2106752277, but had no clue why the test failed on Linux while properly passing on the other platforms as well as my machine and therefore [asked for support](https://github.com/nunit/nunit/pull/4709#issuecomment-2106752277).

But apparently, in the meantime, some change now makes one test fail on all platforms. Still I cannot reproduce the failures on my machine, neither on net462 nor 6.0 nor 8.0:
grafik

I guess I still need some sort of support, or at least a hint on how I can reproduce the tests same as on github

Also note https://github.com/nunit/nunit/pull/4709#issuecomment-2111694347.

**Comment by SeanKilleen** [comment]
Just let me know if https://github.com/nunit/nunit/pull/4709 won't make it, so that I can revert the docs PR and queue it up again.

**Pull request merged**